File: Euclid.Storage/Database/Data/Item/ItemData.cs
using FluentNHibernate.Mapping;

namespace Euclid.Storage.Database.Data
{
    class ItemMapping : ClassMap<ItemData>
    {
        public ItemMapping()
        {
            Table("item");
            Id(x => x.Id, "id").GeneratedBy.Identity();
            Map(x => x.OrderId, "order_id");
            Map(x => x.OwnerId, "user_id");
            Map(x => x.RoomId, "room_id");
            Map(x => x.DefinitionId, "definition_id");
            Map(x => x.X, "x").Generated.Insert();
            Map(x => x.Y, "y").Generated.Insert();
            Map(x => x.Z, "z").Generated.Insert();
            Map(x => x.WallPosition, "wall_position").Generated.Insert();
            Map(x => x.Rotation, "rotation").Generated.Insert();
            Map(x => x.CustomData, "custom_data");

            References(x => x.OwnerData, "user_id")
                .ReadOnly()
                .NotFound.Ignore();
        }
    }

    public class ItemData
    {
        public virtual int Id { get; set; }
        public virtual int OrderId { get; set; }
        public virtual int OwnerId { get; set; }
        public virtual PlayerData OwnerData { get; set; }
        public virtual int RoomId { get; set; }
        public virtual int DefinitionId { get; set; }
        public virtual int X { get; set; }
        public virtual int Y { get; set; }
        public virtual double Z { get; set; }
        public virtual string WallPosition { get; set; }
        public virtual int Rotation { get; set; }
        public virtual string CustomData { get; set; }
    }
}

File: euclid-storage/src/main/java/euclid/storage/database/data/ItemData.java
package euclid.storage.database.data;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;

@Entity
@Table(name = "item")
public class ItemData {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private int id;

    @Column(name = "order_id")
    private int orderId;

    @Column(name = "user_id")
    private int ownerId;

    @ManyToOne(fetch = FetchType.LAZY, optional = true)
    @JoinColumn(name = "user_id", insertable = false, updatable = false)
    private PlayerData ownerData;

    @Column(name = "room_id")
    private int roomId;

    @Column(name = "definition_id")
    private int definitionId;

    @Column(name = "x", insertable = false)
    private int x;

    @Column(name = "y", insertable = false)
    private int y;

    @Column(name = "z", insertable = false)
    private double z;

    @Column(name = "wall_position", insertable = false)
    private String wallPosition;

    @Column(name = "rotation", insertable = false)
    private int rotation;

    @Column(name = "custom_data")
    private String customData;

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getOrderId() {
        return orderId;
    }

    public void setOrderId(int orderId) {
        this.orderId = orderId;
    }

    public int getOwnerId() {
        return ownerId;
    }

    public void setOwnerId(int ownerId) {
        this.ownerId = ownerId;
    }

    public PlayerData getOwnerData() {
        return ownerData;
    }

    public void setOwnerData(PlayerData ownerData) {
        this.ownerData = ownerData;
    }

    public int getRoomId() {
        return roomId;
    }

    public void setRoomId(int roomId) {
        this.roomId = roomId;
    }

    public int getDefinitionId() {
        return definitionId;
    }

    public void setDefinitionId(int definitionId) {
        this.definitionId = definitionId;
    }

    public int getX() {
        return x;
    }

    public void setX(int x) {
        this.x = x;
    }

    public int getY() {
        return y;
    }

    public void setY(int y) {
        this.y = y;
    }

    public double getZ() {
        return z;
    }

    public void setZ(double z) {
        this.z = z;
    }

    public String getWallPosition() {
        return wallPosition;
    }

    public void setWallPosition(String wallPosition) {
        this.wallPosition = wallPosition;
    }

    public int getRotation() {
        return rotation;
    }

    public void setRotation(int rotation) {
        this.rotation = rotation;
    }

    public String getCustomData() {
        return customData;
    }

    public void setCustomData(String customData) {
        this.customData = customData;
    }
}
